src/main.cpp

```diff
@@ -2219,6 +2223,14 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
             pindex->nStatus |= BLOCK_HAVE_UNDO;
         }
 
+        // Now that all consensus rules have been validated, set nBranchId
```

Comment:
What if there are consensus rules that depend on the branch ID?

Reply:
See my earlier comment: the branch ID set in the block index is only being cached for performance reasons, and is only set when we know the block actually satisfies it. All consensus rules that depend on the branch ID must use the height APIs directly.

Reply:
OK, I see there's [already a comment about this](https://github.com/zcash/zcash/pull/2898/files#diff-499d7ee7998a27095063ed7b4dd7c119R145) at the declaration, so I guess that's fine.